ocaml/xapi/extauth_plugin_ADpbis.ml

```diff
@@ -20,6 +20,33 @@ open D
 
 open Stdext.Xstringext
 
+let lwsmd_service = "lwsmd"
+
+let is_ad_enabled ~__context =
+  Helpers.get_localhost ~__context
+  |> (fun self -> Db.Host.get_external_auth_type ~__context ~self)
+  |> (fun x -> x = Xapi_globs.auth_type_AD_Likewise)
+
+let start_lwsmd ~__context =
+  (* if current host is auth with AD, then start lwsmd *)
+  try
+    ignore (Forkhelpers.execute_command_get_output !Xapi_globs.systemctl ["start"; lwsmd_service]);
+    Thread.delay !Xapi_globs.lwsmd_service_bootup_time;
```

Comment:
We don't know after the delay whether the service has started or not. forkexechelpers has a function to gather the status of a systemd service to check this and fail accordingly.

Consider as well adding a maximum wait time after which we consider the service to have failed if it hasn't booted up yet.

Reply:
:+1: 
do you mean https://github.com/xapi-project/forkexecd/blob/master/lib/fe_systemctl.mli#L32

Reply:
Correct